<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FPFConcurrencySection</class>
 <widget class="QWidget" name="FPFConcurrencySection">
  <property name="windowTitle">
   <string>FPF Concurrency</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayoutRoot">
   <item>
    <widget class="QGroupBox" name="groupFPFConcurrency">
     <property name="title">
      <string>FPF Concurrency (Global)</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayoutGlobal">
      <item row="0" column="0">
       <widget class="QLabel" name="labelEnable">
        <property name="text">
         <string>Enable</string>
        </property>
        <property name="toolTip">
         <string>Enable or disable concurrency controls at runtime.</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="checkConcurrencyEnabled">
        <property name="toolTip">
         <string>When checked, concurrency limits are enforced.</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>

      <item row="1" column="0">
       <widget class="QLabel" name="labelGlobalMax">
        <property name="text">
         <string>Global Max Concurrency</string>
        </property>
        <property name="toolTip">
         <string>Upper bound on total concurrent tasks across all providers.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="layoutGlobalMaxConcurrency">
        <item>
         <widget class="QLabel" name="labelGlobalMaxConcurrencyMin">
          <property name="text">
           <string>1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="sliderGlobalMaxConcurrency">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>64</number>
          </property>
          <property name="value">
           <number>12</number>
          </property>
          <property name="toolTip">
           <string>Total number of concurrent FPF operations allowed globally.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelGlobalMaxConcurrencyMax">
          <property name="text">
           <string>12 / 64</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>

      <item row="2" column="0">
       <widget class="QLabel" name="labelGlobalQps">
        <property name="text">
         <string>Global QPS</string>
        </property>
        <property name="toolTip">
         <string>Global queries per second. All runs are staggered by at least 1/QPS seconds; burst is disabled.</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="layoutGlobalQPS">
        <item>
         <widget class="QLabel" name="labelGlobalQPSMin">
          <property name="text">
           <string>0.1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="sliderGlobalQPS">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
          <property name="toolTip">
           <string>Global rate (queries per second). Slider value / 10.0 = QPS (e.g., 20 => 2.0 QPS).</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelGlobalQPSMax">
          <property name="text">
           <string>2.0 / 20.0</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
